### PR TITLE
ES7 Rollout commands

### PR DESCRIPTION
**Pull request by sravfeyn**
I have tested all of these on staging.

@millerdev @esoergel @snopoke

### CODE REVIEW

File: corehq/apps/hqcase/management/commands/reprocess_form_case_es_deletes.py
Comment (sravfeyn) [inline corehq/apps/hqcase/management/commands/reprocess_form_case_es_deletes.py:14]: It will be nice to get a second eye on this logic that decides which pillows we need to reprocess the deletes on. 

The goal of this command is to make sure the deletes in Form/Case ES indices get processed because these deletes are not handled in the reindex command which only copies data across but doesn't delete the data. The logic I followed is this
1. Get a list of active pillows that update form/case ES 
2. Different envs may use different pillows, so an active pillow is inferred based on its checkpoints last modification time.
3. Only deletion part of the pillow.process_change is triggered based on whether a pillow processor is ElasticProcessor or not

File: corehq/apps/hqcase/management/commands/reprocess_form_case_es_deletes.py
Comment (sravfeyn) [inline corehq/apps/hqcase/management/commands/reprocess_form_case_es_deletes.py:29]: Intentionally not adding custom ICDS only local pillows, since ICDS doesn't have a workflow of deleting forms/cases. Other envs don't have any custom local pillows, so using the standard pillows.

File: corehq/apps/hqcase/management/commands/reindex_es_from_remote.py
Comment (sravfeyn) [reply corehq/apps/hqcase/management/commands/reindex_es_from_remote.py:212]: Nice! Thanks

File: corehq/apps/hqcase/management/commands/reindex_es_from_remote.py
Comment (sravfeyn) [reply corehq/apps/hqcase/management/commands/reindex_es_from_remote.py:93]: ?

File: corehq/apps/hqcase/management/commands/initialize_es_indices.py
Comment (sravfeyn) [reply corehq/apps/hqcase/management/commands/initialize_es_indices.py:41]: 👍

File: corehq/apps/hqcase/management/commands/reindex_es_from_remote.py
Comment (sravfeyn) [reply corehq/apps/hqcase/management/commands/reindex_es_from_remote.py:89]: It is possible, but rather unlikely if one follows the rollout process. This command doesn't intend to test/guarantee data integrity.

File: corehq/apps/hqcase/management/commands/reindex_es_from_remote.py
Comment (sravfeyn) [reply corehq/apps/hqcase/management/commands/reindex_es_from_remote.py:160]: The result doesn't seem to contain `completed`. Below is a sample result for the task
```
{'node': 'L5MAj36NQYOdepD0Qrh1BQ', 'id': 22269, 'type': 'transport', 'action': 'indices:data/write/reindex', 'status': {'total': 54, 'updated': 54, 'created': 0, 'deleted': 0, 'batches': 1, 'version_conflicts': 0, 'noops': 0, 'retries': {'bulk': 0, 'search': 0}, 'throttled_millis': 0, 'requests_per_second': -1.0, 'throttled_until_millis': 0}, 'description': 'reindex from [host=localhost port=5200 query={
  "match_all" : {
    "boost" : 1.0
  }}][xforms_2016-07-07] to [xforms_2016-07-07][_doc]', 'start_time_in_millis': 1599049858036, 'running_time_in_nanos': 132674323, 'cancellable': True, 'headers': {}}
```

File: corehq/apps/hqcase/management/commands/reindex_es_from_remote.py
Comment (sravfeyn) [reply corehq/apps/hqcase/management/commands/reindex_es_from_remote.py:154]: Sure

File: corehq/apps/hqcase/management/commands/reindex_es_from_remote.py
Comment (millerdev) [reply corehq/apps/hqcase/management/commands/reindex_es_from_remote.py:158]: Is your example result from a completed task? I think the field is not present until the task has completed.

**Comment by sravfeyn** [comment]
@millerdev Bumping for review.

File: corehq/apps/hqcase/management/commands/reindex_es_from_remote.py
Comment (sravfeyn) [reply corehq/apps/hqcase/management/commands/reindex_es_from_remote.py:158]: Updated now, thanks!

File: corehq/apps/hqcase/management/commands/initialize_es_indices.py
Comment (millerdev) [reply corehq/apps/hqcase/management/commands/initialize_es_indices.py]: It doesn't look like you applied my suggestion, and the change you made did not change the way the command works.

https://github.com/dimagi/commcare-hq/blob/3650eb1652fa2b5ab717ce0ac7c9c67302173921/corehq/apps/hqcase/management/commands/initialize_es_indices.py#L41

File: corehq/apps/hqcase/management/commands/reindex_es_from_remote.py
Comment (millerdev) [reply corehq/apps/hqcase/management/commands/reindex_es_from_remote.py:158]: I don't see what changed. Did you forget to push a commit?

**Review by millerdev — COMMENTED** [review]
There are still a lot of lint errors, some of which will result in runtime errors if not fixed (undefined names, for example).
```
reindex_es_from_remote.py:110:77: E226 missing whitespace around arithmetic operator
reindex_es_from_remote.py:128:79: W291 trailing whitespace
reindex_es_from_remote.py:137:57: W291 trailing whitespace
reindex_es_from_remote.py:138:63: W291 trailing whitespace
reindex_es_from_remote.py:150:9: F841 local variable 'node_id' is assigned to but never used
reindex_es_from_remote.py:170:47: F821 undefined name 'query'
reindex_es_from_remote.py:171:93: F821 undefined name 'query'
reindex_es_from_remote.py:174:83: W291 trailing whitespace
reindex_es_from_remote.py:209:73: E231 missing whitespace after ':'
reprocess_form_case_es_deletes.py:2:1: F401 'django.core.management.CommandError' imported but unused
reprocess_form_case_es_deletes.py:5:1: F401 'pillowtop.get_all_pillow_instances' imported but unused
reprocess_form_case_es_deletes.py:12:51: W291 trailing whitespace
reprocess_form_case_es_deletes.py:83:33: E225 missing whitespace around operator
reprocess_form_case_es_deletes.py:83:34: W291 trailing whitespace
```

**Comment by sravfeyn** [comment]
I think I forgot to push. It's updated now.

File: corehq/apps/hqcase/management/commands/reindex_es_from_remote.py
Comment (millerdev) [reply corehq/apps/hqcase/management/commands/reindex_es_from_remote.py:171]: > F821 undefined name 'query'

This is a critical issue that should be fixed.

**Review by millerdev — COMMENTED** [review]
The lint warnings have not been addressed.

File: corehq/apps/hqcase/management/commands/reindex_es_from_remote.py
Comment (sravfeyn) [reply corehq/apps/hqcase/management/commands/reindex_es_from_remote.py]: Yeah, it was a debug thing. Removed.

**Comment by sravfeyn** [comment]
Updated.

**Comment by sravfeyn** [comment]
Bumping @millerdev

File: corehq/apps/hqcase/management/commands/reindex_es_from_remote.py
Comment (millerdev) [reply corehq/apps/hqcase/management/commands/reindex_es_from_remote.py:171]: > F821 undefined name 'query'

This one has not been fixed yet.

File: corehq/apps/hqcase/management/commands/reprocess_form_case_es_deletes.py
Comment (millerdev) [reply corehq/apps/hqcase/management/commands/reprocess_form_case_es_deletes.py:13]: I don't have enough context to evaluate this. Maybe @proteusvacuum could weigh in?

**Review by millerdev — COMMENTED** [review]
You may find a tool like [git-flake8](https://gist.github.com/millerdev/e25014e4b886a5c619419d56b778a423) handy to check the current branch for flake8 errors.

**Comment by millerdev** [comment]
Not sure why this comment did not appear with the review https://github.com/dimagi/commcare-hq/pull/28450/files#r488029312

**Comment by sravfeyn** [comment]
Thanks Daniel. Sorry that was annoying. My flake8 pre-commit hook wasn't working (silently) since it didn't have a chmod +x. I did that now and tested it a commit and it works. Appreciate the extra snipper for running it on a branch. Also ran that and it is clear now.

**Review by millerdev — COMMENTED** [review]
Approved if no response is needed for https://github.com/dimagi/commcare-hq/pull/28450#discussion_r480102205

File: corehq/apps/hqcase/management/commands/reprocess_form_case_es_deletes.py
Comment (sravfeyn) [reply corehq/apps/hqcase/management/commands/reprocess_form_case_es_deletes.py:13]: @calellowitz  or @snopoke ?

Comment (snopoke) [reply corehq/apps/hqcase/management/commands/reprocess_form_case_es_deletes.py:13]: Logic seems OK to me. You could further filter by topic (unless that's happening somewhere else).

Comment (sravfeyn) [reply corehq/apps/hqcase/management/commands/reprocess_form_case_es_deletes.py:13]: All the topics should get processed per this line https://github.com/dimagi/commcare-hq/pull/28450/files#diff-5906407fba48bdc631775d7847e02a3aR81 below

**Comment by sravfeyn** [comment]
@millerdev I am not able to merge without an approval.

File: corehq/apps/hqcase/management/commands/reprocess_form_case_es_deletes.py
Comment (sravfeyn) [reply corehq/apps/hqcase/management/commands/reprocess_form_case_es_deletes.py:13]: Also the model doesn't have a topic column to filter individually on. https://staging.commcarehq.org/admin/hqadmin/historicalpillowcheckpoint/

**Comment by sravfeyn** [comment]
@millerdev @snopoke Can I get an approval if good?

**Review by snopoke — APPROVED** [review]

**Pull request merged**
